### PR TITLE
statistics: fix region flow calculation

### PR DESCRIPTION
**Pull request by jiyingtk**
### What problem does this PR solve? 
Region flow statistics would be inaccurate when transfer leader occurs.

### What is changed and how it works?
Use time interval reported from region heartbeat to calculate flow.

### Check List 

Tests 

 - Manual test

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type RegionInfo struct {
 	readKeys        uint64
 	approximateSize int64
 	approximateKeys int64
+	reportInterval  uint64
```
Comment (nolouch) [inline server/core/region.go]: Can we use  `pdpb.TimeInterval` directly?

Comment (jiyingtk) [reply server/core/region.go]: OK

File: server/statistics/hot_cache.go [hunk 2]
```diff
@@ -77,15 +77,23 @@ func NewHotStoresStats() *HotStoresStats {
...
+		keysPerSec     uint64
+		reportInterval uint64
 
 		isExpiredInStore func(region *core.RegionInfo, storeID uint64) bool
 	)
 
+	reportInterval = region.GetInterval().GetEndTimestamp() - region.GetInterval().GetStartTimestamp()
+
+	// ignores this region flow information if the report time interval is too short or too long.
+	if reportInterval < minHotRegionReportInterval || reportInterval > 3*RegionHeartBeatReportInterval {
```
Comment (rleungx) [inline server/statistics/hot_cache.go]: I'm not sure if this is necessary. /cc @nolouch

Comment (nolouch) [reply server/statistics/hot_cache.go]: It's necessary, but we should consider `isExpiredInStore(region)` in here.

**Pull request merged**